jujugui/static/gui/src/app/user/user.js

```diff
@@ -26,24 +26,35 @@ if (typeof this.jujugui === 'undefined') {
 const User = class User {
 
   constructor(cfg = {}) {
-    this.storage = cfg.storage || sessionStorage;
+    // We pass in these values to make test setup easier.
+    this.sessionStorage = cfg.sessionStorage || sessionStorage;
+    this.localStorage = cfg.localStorage || localStorage;
+    this._DISCHARGE_TOKEN = 'discharge-token';
   }
 
   // TODO get username
 
-  // TODO get/set identity creds
+  /**
+   Macaroon tokens from the identity manager.
+   */
+  get identity() {
+    return this.localStorage.getItem(this._DISCHARGE_TOKEN);
+  }
+  set identity(token) {
+    this.localStorage.setItem(this._DISCHARGE_TOKEN, token);
+  }
 
   // TODO get/set charmstore creds
 
   /**
    Credentials for the controller connection for the GUI. As a getter,
    it adds convenience attributes to the credentials for handling login flow.
-  /*
-  /* XXX 'controller' isn't quite right--this handles both controller and model
-   connection credentials. Perhaps juju? */
+  */
+  // XXX 'controller' isn't quite right--this handles both controller and model
```

Comment:
I guess this XXX will go away with your next branch, in which we have both controller and model properties.